tools/roctx.py

```diff
@@ -0,0 +1,248 @@
+        try:
+            max_per_name.append(max(list))
+        except:
+            max_per_name.append(0)
+
+    min_per_name = []
+    for list in list_times_per_names:
+        try:
+            min_per_name.append(min(list))
+        except:
+            min_per_name.append(0)
+
+    max_index_per_name = []
+    for list in list_times_per_names:
+        try:
+            max_index_per_name.append(list.index(max(list)))
+        except:
+            max_index_per_name.append(0)
+
+    max_occur_per_name = []
+    for list in list_times_per_names:
+        try:
+            max_occur_per_name.append(list.count(max(list)))
+        except:
+            max_occur_per_name.append(0)
+
+    total_time = sum(sum_per_name)
+
+    d = {
+        'SUM': sum_per_name,
+        'MIN': min_per_name,
+        'MAX': max_per_name,
+        'MAX_INDEX': max_index_per_name,
+        'MAX_OCCUR': max_occur_per_name
+    }
+    df2 = pd.DataFrame(d)
+    df2.index = list_names
+    df2.sort_values(by=['SUM'], inplace=True, ascending=False)
+
+    if (args.debug):
+        print(df2)
+        print("\nTOTAL TIME: %s us" % total_time)
+    return df2, total_time
+
+
+def run():
+    args = parse_args()
+    onnx_path = args.onnx_file
+    repeat_count = args.repeat
+    if (repeat_count == 0 or repeat_count == float('inf') or not repeat_count):
+        raise Exception(
+            "Repeat count is either, 0, infinity or not defined. Quitting.")
+    migraphx_args = args.migraphx_args
+    if not (onnx_path):
+        raise Exception("No ONNX file is provided to run.")
+    onnx_rpath = os.path.realpath(onnx_path)
+    print(onnx_rpath)
+    #configurations
+    configs = '--hip-trace --roctx-trace --flush-rate 10ms --timestamp on'
+    output_dir = '-d %s' % args.out
+    executable = '/opt/rocm/bin/migraphx-driver roctx %s %s' % (onnx_rpath,
+                                                                migraphx_args)
+    process_args = configs + ' ' + output_dir + ' ' + executable
+    for i in range(repeat_count):
+        os.system('rocprof ' + process_args)
+    print("RUN COMPLETE.")
+
+
+def main():
+
+    args = parse_args()
+    print(args)
+    file = args.json_path
+
+    if (args.study_name):
+        filename = args.study_name + ".csv"
+    else:
+        filename = "output" + datetime.now().strftime(
+            "%Y_%m_%d-%I:%M:%S_%p") + ".csv"
+
+    with open(filename, 'a') as f:
+        f.write(args.onnx_file)
+        f.write('\n')
+        f.write(args.migraphx_args)
+        f.write('\n')
+
+    if (args.run):
+        curr = os.path.abspath(os.getcwd())
+        if not os.path.exists('/tmp/rocmProfileData'):
+            print("rocmProfileData does not exist. Cloning.")
+            os.system(
+                'git clone https://github.com/ROCmSoftwarePlatform/rocmProfileData.git /tmp/rocmProfileData'
+            )
+
+        os.chdir("/tmp/rocmProfileData/rocpd_python/")
+        os.system('python setup.py install')
```

Comment:
Yes it will change the users package, though this package is not supposed to change anything but just add this package in the global python packages, as far as I can tell, considering what it does:
```
from setuptools import setup, find_packages

setup(name = 'rocpd',
      version = '1.0',
      description = 'RocmProfileData profiling format',
      #packages = find_packages(),
      packages = { 'rocpd' },
      include_package_data=True, 
      python_requires='>=3.6',
)
```